File: analytics/src/SensorEvent.ts
export interface SensorEvent {
  timestamp: string;
  type: string;
  device: string;
  measurement: string;
  message: string;
  statisticData?: {
    // Used for numeric measurements
    currentValue: number;
    deviationNominal: number;
    deviationPercent: number;
    runningAverage: number;
    historyLength: number;
  };
}

export enum SensorEventType {
  NUMERIC = "NUMERIC",
  BINARY = "BINARY",
}

File: EventInfo/Models/SensorEvent.cs
namespace EventInfo.Models;

public class SensorEvent
{
    public DateTime Timestamp { get; set; }
    public SensorEventType Type { get; set; }
    public string? Device { get; set; }
    public string? Measurement { get; set; }
    public string? Message { get; set; }
    public StatisticData? StatisticData { get; set; }
}

public class StatisticData
{
    public double CurrentValue { get; set; }
    public double DeviationNominal { get; set; }
    public double DeviationPercent { get; set; }
    public double RunningAverage { get; set; }
    public int HistoryLength { get; set; }
}

public enum SensorEventType
{
    NUMERIC,
    BINARY
}
